src/main/java/Parser.java

```diff
@@ -0,0 +1,122 @@
+                            }
+                        }
+                    }
+                    Task task = new Deadline(sd, LocalDate.parse(sb));
+                    duke.getTasks().addTask(task);
+                    duke.getStorage().writeFile(duke.getTasks().getTaskList());
+                    return duke.getUi().printAddTask(task, duke.getTasks());
+                } else if (strArr[0].equals("event")) {
+                    String sd = "";
+                    String sa = "";
+                    boolean b = true;
+                    for (int i = 1; i < strArr.length - 1; i++) {
+                        if (strArr[i].equals("/at")) {
+                            b = false;
+                            sd = sd.substring(0, sd.length() - 1);
+                        }
+                        if (b) {
+                            sd += strArr[i] + " ";
+                        } else {
+                            if (i == strArr.length - 2) {
+                                sa = sa + strArr[i + 1];
+                            } else {
+                                sa = sa + strArr[i + 1] + " ";
+                            }
+                        }
+                    }
+
+                    Task task = new Event(sd, LocalDate.parse(sa));
+                    duke.getTasks().addTask(task);
+                    duke.getStorage().writeFile(duke.getTasks().getTaskList());
+                    return duke.getUi().printAddTask(task, duke.getTasks());
+                } else if (strArr[0].equals("find")) {
+                    String keyword = strArr[1];
+                    TaskList taskLst = new TaskList();
+                    for (int i = 0; i < duke.getTasks().getSize(); i++) {
+                        if (duke.getTasks().getTask(i).getDescription().contains(keyword)) {
+                            taskLst.addTask(duke.getTasks().getTask(i));
+                        }
+                    }
+                    return duke.getUi().printMatchingTasks(taskLst);
+                }
+            }
+        } catch (Exception e) {
+            return duke.getUi().showError(e.toString());
+        }
+        return "";
+    }
+
+    /**
+     * Checks whether has invalid user inputs.
+     * @param arr Array of user input.
+     * @throws DukeException Exception specific to Duke.
+     */
+    public void check(String[] arr) throws DukeException {
+        if (arr.length == 1 && (arr[0].equals("todo") || arr[0].equals("deadline") || arr[0].equals("event") ||
+                arr[0].equals("done") || arr[0].equals("delete") || arr[0].equals("find"))) {
+            throw new DukeException(arr[0]);
+        } else if (!arr[0].equals("todo") && !arr[0].equals("deadline") && !arr[0].equals("event") &&
+                !arr[0].equals("list") && !arr[0].equals("bye") && !arr[0].equals("done") 
+                && !arr[0].equals("delete") && !arr[0].equals("find")) {
+            throw new DukeException("other");
```

Comment:
Readiblity is quite poor here :(

Switching the if else checks around would allow you to skip `&& (arr[0].equals("todo") || arr[0].equals("deadline") || arr[0].equals("event") ||
                arr[0].equals("done") || arr[0].equals("delete") || arr[0].equals("find"))` just checking for length instead. 

So:
if statement 1) command I don't understand? (ignoring description) 
if statement 2) command without description? (all commands here would be understood) 

Instead of:
if statement 1) command without description? && command I understand?
if statement 2) command with description but I don't understand?

Comments would be nice to describe what is being checked at each if statement.

Do also consider using the logic under `public String parse(String userInput, Duke duke)` to avoid doing checks on the command string twice to keep maintaining simple (ie don't have to update 2 places when commands change).